life_expectancy/main.py

```diff
@@ -1,33 +1,45 @@
 """This is the main module"""
 from pathlib import Path
-
 import argparse
 import pandas as pd
-
-from life_expectancy.cleaning import clean_data
-from life_expectancy.load_save_data import load_data, save_data
-
+from life_expectancy.cleaning import filter_data, clean_data, feature_cleaning
+from life_expectancy.load_save_data import TSVFileReader, JSONFileReader, FileProcessor, save_data
+from life_expectancy.countries import Region
 
 script_dir = Path(__file__).resolve().parent
-input_file_path = script_dir/"data"/"eu_life_expectancy_raw.tsv"
+input_file_path_tsv= script_dir/"data"/"eu_life_expectancy_raw.tsv"
+input_file_path_json = script_dir/"data"/"eurostat_life_expect.zip"
 output_file_path = script_dir/"data"/"pt_life_expectancy.csv"
 
 
-def main(region: str = 'PT') -> pd.DataFrame:
+def main(region: Region = Region.PT) -> pd.DataFrame:
     """
     Main function that load, cleans and save a cleaned version of the life expectancy data
     Args:
-        region (str): String with region information
+        region (Regions): Enum with region information
     """
-    life_exp_raw_data = load_data(input_file_path)
-    life_exp = clean_data(life_exp_raw_data, region)
+    # Verify if the region is in the list of Contries and Regions
+    if region.value not in Region.__members__:
+        raise ValueError("This is an Invalid Country")
+
+    file_reader = JSONFileReader()
+    file_processor = FileProcessor(file_reader)
+    life_exp_raw_data = file_processor.processor_file(input_file_path_json)
+
+    if isinstance(file_reader, TSVFileReader):
+        life_exp = clean_data(life_exp_raw_data)
+    else:
+        life_exp = feature_cleaning(life_exp_raw_data)
```

Comment:
**Question**: I don't understand this logic... why are you checking if file_reader is a TSVFileReader when you have already declared it as a JSONFileReader?

Reply:
A possible solution to your conundrum, would be to add both reading and cleaning behaviors' to the same class (and then renaming it from JSONFileReader to something more descriptive). That way, your choice or strategy will already determine _both_ cleaning and loading

Reply:
yes, I was just trying that the only change between a JSONFileReader and TSVFileReader was just replacing JSONFileReader by TSVFileReader and vice-versa. But I have change the name of the class to FileHandlerStrategy (with two subclasses JSONFileHandler and TSVFileHandler), with two methods, one to load and the other to clean. Thanks :)